src/StatusLine.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot;
+
+use Teapot\StatusCodeException\InvalidStatusCodeException;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class StatusLine
```

Comment:
@shrikeh 

I usually make all value objects `final` to maintain immutability - if doing so, it's just natural to extract an interface because it's impossible to mock `final` classes.